web/modules/custom/dpl_patron_page/src/Plugin/Block/PatronPageBlock.php

```diff
@@ -0,0 +1,166 @@
+
+  /**
+   * Drupal config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  private ConfigFactoryInterface $configFactory;
+
+  /**
+   * PatronPageBlock constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   Drupal config factory to get FBS and Publizon settings.
+   * @param \Drupal\dpl_library_agency\BranchSettings $branchSettings
+   *   Branch settings.
+   * @param \Drupal\dpl_library_agency\Branch\BranchRepositoryInterface $branchRepository
+   *   Branch repository.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $configFactory, private BranchSettings $branchSettings, private BranchRepositoryInterface $branchRepository) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configuration = $configuration;
+    $this->configFactory = $configFactory;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+      $container->get('dpl_library_agency.branch_settings'),
+      $container->get('dpl_library_agency.branch.repository'),
+    );
+  }
+
+  /**
+   * Checks whether the library has enabled text messages.
+   *
+   * @return string
+   *   True if enabled, false if disabled.
+   */
+  private function textNotificationsEnabled(): string {
+    $patron_page_settings = $this->configFactory->get('patron_page.settings');
+    return empty($patron_page_settings->get('text_notifications_enabled')) ? 'false' : 'true';
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return mixed[]
+   *   The app render array.
+   *
+   * @throws \Safe\Exceptions\JsonException
+   */
+  public function build() {
+    $patron_page_settings = $this->configFactory->get('patron_page.settings');
+    $general_config = $this->configFactory->get('dpl_library_agency.general_settings');
+
+    $dateConfig = $general_config->get('pause_reservation_start_date_config');
+    if (is_null($dateConfig)) {
+      $dateConfig = "";
+    }
+
+    $data = [
+      // Configuration.
+      'text-notifications-enabled-config' => $this->textNotificationsEnabled(),
+      'blacklisted-pickup-branches-config' => DplReactAppsController::buildBranchesListProp($this->branchSettings->getExcludedReservationBranches()),
+      'branches-config' => DplReactAppsController::buildBranchesJsonProp($this->branchRepository->getBranches()),
```

Comment:
This should really be a service. But I understand that you stick for this for now.
Consider writing a comment about it.